doc, CLI UX and best practices improved

faelmori opened the pull request:
I've synced the fork with your latest updates and made a big merge with all the changes I made (including a few extra ones to improve the CLI UX). Sorry for the inconvenience and lots of pull requests. 


It won't happen again.

Ex:

from here: 
![image](https://github.com/user-attachments/assets/2cd72a1f-30b9-41a7-bae7-5ed8feddeb00)


to here:
![image](https://github.com/user-attachments/assets/cedd3e6c-1412-47c5-a2bf-8a4e5c129ac3)


And the other things... See ya!

mms-gianni commented:
@faelmori Does this include your previous changes? 

Sorry. Im currently very busy. But I'm very thankful for your contributions!

faelmori commented:
Yes man!

All typo corrections; improved variable names and structure definitions; improved and corrected structure definitions as well; updated all dependencies to mitigate known vulnerabilities; improved UX in the CLI with different colors for different CLI elements, for example: root commands are one color, titles another, flags another, etc...); improved README.md with a more professional and attractive design with badges, summary, more colors, etc;

I inserted a tutorial of the process to build directly from source using UPX to reduce the size of the binary, tag and git description for the internal version of the binary, I used some flags in the binary build to make it more secure (removing paths from the builder's machine, obfuscating user profile data that ends up in the binary, etc.).

faelmori commented:
> @faelmori Does this include your previous changes?
> 
> Sorry. Im currently very busy. But I'm very thankful for your contributions!

Are you discontinuing this CLI? It must have been over a month since I saw any movement on it, including a review of my pull request. That's fine with me, I'm just asking to know if it's worth continuing to code in it if you don't intend to use it, etc. Maybe you already have a more elegant solution as a CLI, I don't know... lol

mms-gianni commented:
No. 100% not!

Im currently working on Kubero V3 . This might change some routes slightly. But it will be feature matching.

faelmori commented:
> No. 100% not!
> 
> Im currently working on Kubero V3 . This might change some routes slightly. But it will be feature matching.

COOL! Good to know..Then I will continue contributing! See you later!

mms-gianni reviewed (approved):
Wow. Finally found some time to review it. A little embarassing that i made so many typos. 

Thanks a lot for your contributions. @faelmori

faelmori commented:
YEAAH! Thanks man! I hope I have helped and I hope to help a lot more! Count on me in the near future, as there may be new things coming from here too. Always forward!
